src/incoming-message-stream.ts

```diff
@@ -49,7 +49,7 @@ class IncomingMessageStream extends Transform {
       // Get the full packet length
       const length = this.bl.readUInt16BE(2);
 
-      if (this.bl.length >= length) {
+      if (this.bl.length >= length && length >= HEADER_LENGTH) {
```

Comment:
I'm wondering if it would make sense to call `callback` with an error here, if `length` is either smaller than `HEADER_LENGTH` or larger than the previously negotiated packet length? 🤔

Reply:
I don't think we need a callback with an error at this place.

Reply:
The stream API expects errors to be returned back via the `callback` that is passed to the `_transform` method. This callback is passed from `_transform` down to `processBufferedData`.

If the length we read from the packet is invalid, this means the whole stream is "corrupted" and the connection should be shut down and for sure can't be used anymore. The proper way to signal this back is to call `callback` with an `Error` parameter and return from  any further processing in `processBufferedData`. This way the error gets bubbled up to wherever the `IncomingMessageStream` is used.

Optimally, this should then cause the connection to be closed properly, but I'm not sure if that's going to happen without testing this properly.

With the fix you put in place here, the initial error you encountered is no longer happening (so the Node process is not going to crash), but I think this is silently corrupting the connection and making it stuck, until the underlying socket times out or the server side closes the connection forcefully.

Does this make sense? 🙇‍♂️

Reply:
Making sense now. I had to manually close the connection on the server side.